### PR TITLE
Remove the use of `std::memset`

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

`std::memset` was being used to zero-initialize some C structs so I replaced that with their modern C++ equivalent. In some cases, no extra initialization code was required since functions like `make_unique` already ensure the members get initialized.

https://en.cppreference.com/w/cpp/language/zero_initialization

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Audio/SoundFileReaderMp3.cpp [hunk 1]
```diff
@@ -99,8 +99,8 @@ bool SoundFileReaderMp3::check(InputStream& stream)
 ////////////////////////////////////////////////////////////
 SoundFileReaderMp3::SoundFileReaderMp3() : m_numSamples(0), m_position(0)
 {
-    std::memset(&m_io, 0, sizeof(m_io));
-    std::memset(&m_decoder, 0, sizeof(m_decoder));
+    m_io      = {};
+    m_decoder = {};
```
Comment (Bromeon) [inline src/SFML/Audio/SoundFileReaderMp3.cpp:103]: Could we not use the constructor initializer list here?

Comment (ChrisThrasher) [reply src/SFML/Audio/SoundFileReaderMp3.cpp]: Good catch! I fixed that.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/Unix/JoystickImpl.cpp [hunk 2]
```diff
@@ -398,8 +398,7 @@ std::string getJoystickName(unsigned int index)
     if (fd >= 0)
     {
         // Get the name
-        char name[128];
-        std::memset(name, 0, sizeof(name));
+        char name[128] = {0};
```
Comment (kimci86) [inline src/SFML/Window/Unix/JoystickImpl.cpp:401]: This could be simply `char name[128] = {};` or `char name[128] = "";`.

Comment (ChrisThrasher) [reply src/SFML/Window/Unix/JoystickImpl.cpp:401]: Fixed, thanks

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by JonnyPtn — APPROVED** [review]

**Review by Bambo-Borris — APPROVED** [review]

**Review by eXpl0it3r — APPROVED** [review]

File: src/SFML/Audio/SoundFileReaderMp3.cpp (hunk 1 above)
Comment (eXpl0it3r) [reply src/SFML/Audio/SoundFileReaderMp3.cpp]: @Bromeon does the fix match your expectations? 😄

**Comment by eXpl0it3r** [comment]
Thanks for the 🧹🪣 🙂

**Pull request merged**